lib/train/file.rb

```diff
@@ -95,77 +74,65 @@ def source
       end
     end
 
-    def pipe?
-      type == :pipe
-    end
-
-    def mode?(sth)
-      mode == sth
-    end
-
-    def owned_by?(sth)
-      owner == sth
-    end
-
-    def grouped_into?(sth)
-      group == sth
+    def source_path
+      @path
     end
 
-    def linked_to?(dst)
-      link_path == dst
+    # TODO: This methods needs to raise error if not implemented currently it set to nil since conversion
```

Comment:
I don't think this is a "TODO" - this is a design choice. Since only Windows will populate this field and it's nil by default, this is a totally acceptable implementation. I think the comment should read something like:

```
# product_version is primarily used by Windows operating systems only and will be overwritten
# in Windows-related classes. Since this field is returned for all file objects, the acceptable
# default value is nil
```